### PR TITLE
Add PrivateEndpointConnection Swagger

### PR DESCRIPTION
**Pull request by beoberha**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by daschult** [comment]
@openapibot sdkautomation rebuild

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/PrivateLinkResources.json [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+      "description": "ARM proxy resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {}
+    },
+    "PrivateLink": {
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/PrivateLinkResources.json]: Please check with Mario whether this should be PrivateLink vs PrivateLinkResource

File: specification/sql/resource-manager/readme.md [hunk 2]
```diff
@@ -214,6 +214,8 @@ input-file:
 - Microsoft.Sql/preview/2018-06-01-preview/managedInstances.json
 - Microsoft.Sql/preview/2018-06-01-preview/FailoverDatabases.json
 - Microsoft.Sql/preview/2018-06-01-preview/FailoverElasticPools.json
+- Microsoft.Sql/preview/2018-06-01-preview/PrivateEndpointConnections.json
```
Comment (jaredmoo) [inline specification/sql/resource-manager/readme.md]: Please also add to package-composite-v3

**Review by jaredmoo — CHANGES_REQUESTED** [review]

File: specification/sql/resource-manager/readme.md (hunk 2 above)
Comment (beoberha) [reply specification/sql/resource-manager/readme.md]: Fixed

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/PrivateLinkResources.json (hunk 1 above)
Comment (beoberha) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/PrivateLinkResources.json]: Discussed with Bo from NRP and he said we should use PrivateLinkResource. Fixed in most recent commit.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/PrivateLinkResources.json [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+        "x-ms-examples": {
+          "Gets private link resources for SQL.": {
+            "$ref": "./examples/PrivateLinkResourcesGet.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "PrivateLinkListResult": {
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/PrivateLinkResources.json]: PrivateLinkResourceListResult

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/examples/PrivateEndpointConnectionUpdate.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    "subscriptionId": "00000000-1111-2222-3333-444444444444",
+    "resourceGroupName": "Default",
+    "serverName": "test-svr",
+    "privateEndpointConnectionName": "private-endpoint-connection-name",
+    "api-version": "2018-06-01-preview",
+    "parameters": {
+      "properties": {
+        "privateLinkServiceConnectionState": {
+          "status": "Approved",
+          "description":  "Approved by johndoe@company.com"
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/examples/PrivateEndpointConnectionUpdate.json]: You know company.com is real? I don't think they want us to use them as an example ;) Please use known examples such as contoso or example.com. "admin@example.com" would be perfect.

Comment (beoberha) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/examples/PrivateEndpointConnectionUpdate.json]: removed from this PR. Will address when I create PLR PR

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/PrivateLinkResources.json (hunk 3 above)
Comment (beoberha) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/PrivateLinkResources.json]: removed from this PR. Will address when I create PLR PR

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/PrivateEndpointConnections.json [hunk 5]
```diff
@@ -0,0 +1,425 @@
...
+      "description": "Properties of a private endpoint connection.",
+      "type": "object",
+      "properties": {
+        "privateEndpoint": {
+          "$ref": "#/definitions/PrivateEndpointProperty",
+          "description": "Private endpoint which the connection belongs to."
+        },
+        "privateLinkServiceConnectionState": {
+          "$ref": "#/definitions/PrivateLinkServiceConnectionStateProperty",
+          "description": "Connection State of the Private Endpoint Connection."
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/PrivateEndpointConnections.json]: Nit: capitalization is only for products (e.g. "Azure SQL Database"), not product features ("private endpoint connection"). Please find & replace.

**Review by jaredmoo — APPROVED** [review]

**Review by anthony-c-martin — CHANGES_REQUESTED** [review]

**Review by anthony-c-martin — APPROVED** [review]

**Review by sergey-shandar — APPROVED** [review]

**Pull request merged**